llvm/lib/Target/RISCV/RISCVSchedule.td

```diff
@@ -112,6 +112,10 @@ def WriteFST16        : SchedWrite;    // Floating point sp store
 def WriteFST32        : SchedWrite;    // Floating point sp store
 def WriteFST64        : SchedWrite;    // Floating point dp store
 
+// CMOV for sifive-p450.
```

Comment:
This is intrusive, can we just use `InstRW` to override the resources in sifive-p450 schedule model? Or is it possible that we may reuse `WriteCMOV`/`ReadCMOV` someday? For example, `Zbt` is back to `Zb*` extensions.

Reply:
Why is having a `WriteCMOV` and a `ReadCMOV` intrusive? How are these `Sched<Read|Write>` different than any other `Sched<Read|Write>`? I understand that not every CPU may use the CMOV instructions, but this is the case for CPUs that do not use vector instructions.

It is my understanding that if a CPU does not implement vector instructions, then it doesn't need to specify WriteRes and ReadAdvance for vector instructions, as this is the case in the `RocketModel`. It would be the same thing for CMOV instructions.

I think WriteCMOV and ReadCMOV are minimally invasive. It is my goal to use `InstRW` as little as possible in the SchedModels -- SchedRead and SchedWrite aim to replace InstRW in my opinion.

Reply:
Thanks, your thought makes sense to me. My concern is that we won't have CMOV in the near future and we need a `UnsupportedCMOV` for almost all new schedule models (except SiFive's), which is already the situation for `UnsupportedSchedSFB`. This is because SFB and this CMOV are only for vendor extensions/features and my thought/strategy is that we should seperate vendor-specific parts from **standard** parts.
Actually, I think there should be a `RISCVInstrInfoSFB.td` which is split out from `RISCVInstrInfo.td`.